pkgs/by-name/en/ente-web/package.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchYarnDeps
+, nodejs
+, yarnConfigHook
+, yarnBuildHook
+
+}: stdenv.mkDerivation (finalAttrs: rec {
+
+
+  pname = "ente-web";
+  version = "photos-v0.9.5";
+
+  src =
+    let
+      repo = fetchFromGitHub {
+        owner = "ente-io";
+        repo = "ente";
+        sparseCheckout = [ "web" ];
+        rev = version;
+        fetchSubmodules = true;
+        hash = "sha256-YJuhdMrgOQW4+LaxEvZNmFZDlFRBmPZot8oUdACdhhE=";
+      };
+    in
+    "${repo}/web";
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${finalAttrs.src}/yarn.lock";
+    hash = "sha256-ZGZkpHZD2LoMIXzpQRAO4Fh9Jf4WxosgykKnn7I1+2g=";
+  };
+
+  nativeBuildInputs = [
+    yarnConfigHook
+    yarnBuildHook
+    nodejs
+  ];
+
+  installPhase = ''
+    cp -r apps/photos/out $out
+  '';
+
+  meta = with lib; {
+    description = "Web client for Ente Photos";
+    homepage = "https://ente.io/";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ surfaceflinger pinpox ];
+    mainProgram = "web";
```

Comment:
I habven't built this locally, but I will notify you later how to adjust the `installPhase` to improve this situation. I also guess that the resulting `$out` has a weird directory layout structure. You may also want to consider using `npmHooks.npmInstallHook`.

Reply:
I had a quick look at `npmHooks.npmInstallHook` but didn't get it to work, let me know if you do. The directory structure as it currently is allows hosting the package using something like nginx (see https://wiki.nixos.org/wiki/Ente_Photos for an example).

Let me know if you find a more elegant way to do this

Reply:
> I had a quick look at `npmHooks.npmInstallHook` but didn't get it to work, let me know if you do. The directory structure as it currently is allows hosting the package using something like nginx (see [wiki.nixos.org/wiki/Ente_Photos](https://wiki.nixos.org/wiki/Ente_Photos) for an example).

So this package doesn't generate an executable? Or does nginx uses this executable? And other html/js files? I think I read somewhere you mentioned you have a module sort of working for this, perhaps it'd be better to add it here.

Reply:
No executable as far as I can tell, just outputs a directory with stuff to put on a webserver. I took inspiration from other packages like zabbix or vaultwarden that have a separate package for the web frontend. I would like to expose it as separate top-level package not tied to the module, since it can make sense to put the front- and backend on different machines.

Reply:
> No executable as far as I can tell, just outputs a directory with stuff to put on a webserver.

Then why did you specify a `meta.mainProgram`?

> I took inspiration from other packages like zabbix or vaultwarden that have a separate package for the web frontend. I would like to expose it as separate top-level package not tied to the module, since it can make sense to put the front- and backend on different machines.

I agree with this approach :+1:.